src/business-logic/services/test/device-service.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { DeviceServiceImplementation } from '../device-service-implementation'
+import { Device, TelemetricsDevice } from '../../../model/entities/device'
+import { mock, when, verify, instance } from '@typestrong/ts-mockito'
+import { MongoDeviceRepository } from '../../../data-access/repositories/mongo/mongo-device-repository'
+import { DeviceService } from '../interfaces/device-service'
+import { DeviceEventEmitter } from '../interfaces/device-event-emitter'
+
+describe('DeviceService', () => {
+  let testDevice: TelemetricsDevice
+  let deviceRepository: MongoDeviceRepository
+  let deviceEventEmitter: DeviceEventEmitter
+  const mockedValue = mock<MongoDeviceRepository>() 
+
+  beforeEach(() => {
+    testDevice = mock<TelemetricsDevice>()
+    deviceRepository = instance(mockedValue)
+    deviceEventEmitter = mock<DeviceEventEmitter>()
+  })
+
+
+  it('should read all configurations', async () => {
+    const devices: Device[] = [testDevice]
+    const mockedService = mock<DeviceService>()
+    const testee: DeviceServiceImplementation = new DeviceServiceImplementation(deviceRepository, deviceEventEmitter)
+    when(mockedService.getDevices()).thenReturn(Promise.resolve(devices))
+    
+    await testee.getDevices()
+
+    verify(mockedValue.getDevices()).called()
+  })
```

Comment:
A few things here:
- You are reading devices, not configurations, so it should be named `should read all devices`.
- A common naming convention for our tests are as follows:
```
describe(DeviceServiceImplementation.name), () => {
  describe(DeviceServiceImplementantion.protoype.getDevices.name, () => {
    it('reads all devices', () => {
      ...
    }
  }

  describe(DeviceServiceImplementation.prototype.create.name, () => {
    // Tests for that method here
  }
}
```
- Your test is testing implementation, not behaviour. I.e. it's verifying that something within the method has been called instead of verifying the result returned by the method.
Here is how I would implement this test:
![image](https://github.com/user-attachments/assets/5718a8c2-f1e8-4ab6-9621-7c01aad61c7c)
Notice how I am not using any "global" mocking. This means I don't have to worry about the integrety of this test when writing other tests. And I can modify other tests to my hearts desire without ever affecting this one :)